mmengine/hooks/early_stopping_hook.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import warnings
+from math import inf, isfinite
+from typing import Optional, Union
+
+from mmengine.registry import HOOKS
+from .hook import Hook
+
+DATA_BATCH = Optional[Union[dict, tuple, list]]
+
+
+@HOOKS.register_module()
+class EarlyStoppingHook(Hook):
```

Comment:
Consider recording the state of early stopping to support resuming training.

Reply:
`MessageHub` will save all history metrics during training, maybe we could utilize it to resume training